wcfsetup/install/files/lib/data/user/UserProfile.class.php

```diff
@@ -939,7 +939,7 @@ public function getRank(): ?UserRank
             return null;
         }
 
-        return UserRankCacheBuilder::getInstance()->getRank($this->rankID);
+        return (new UserRankCache())->getCache()[$this->rankID] ?? null;
```

Comment:
That looks like an opportunity for a `getRank()` helper method on `UserRankCache`?